### PR TITLE
Update workflow for automation

### PR DESCRIPTION
**Pull request by ismaelrumzan**
In this PR, we are applying an overlay after pulling the openapi spec from https://openapi.vercel.sh/
This overlay fixes all the errrors that the speakeasy lint identifies when it runs, which stops it from being able to generate the SDK
- Pull source from URL
- Add lint to update `path-params` rule to warning only
- Add overlay to fix lint errors

How to test
- Clone the repo: https://github.com/vercel/vercel
- Change to this branch and make a copy `git checkout -b ....`
- `pnpm i` on the root should install speakeasy - otherwise https://www.speakeasy.com/docs/speakeasy-cli/getting-started
- CD into `packages/SDK` and do `speakeasy run`
- It should run with no errors (pass the lint) and then create a bunch of changes to update the SDK (this is why we make a new branch as we don't want to run speakeasy on this branch - once this merges, we will run it with the GH action)

### CODE REVIEW

File: packages/sdk/.speakeasy/workflow.yaml [hunk 1]
```diff
@@ -3,7 +3,10 @@ speakeasyVersion: latest
 sources:
     vercel-OAS:
         inputs:
-            - location: ../../open-api.yaml
+            - location: https://openapi.vercel.sh/
+        overlays:
+            - location: ../../overlay.yaml
+        ruleset: speakeasyBarRuleset
```
Comment (simplesagar) [inline packages/sdk/.speakeasy/workflow.yaml]:
```suggestion
        ruleset: barRuleset
```

File: packages/sdk/.speakeasy/lint.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+lintVersion: 1.0.0
+defaultRuleset: speakeasyBarRuleset
+rulesets:
+ barRuleset:
```
Comment (simplesagar) [inline packages/sdk/.speakeasy/lint.yaml]: you may want to choose a ruleset name specific to Vercel ?

**Review by simplesagar — COMMENTED** [review]
Minor comment on the naming of the custom linting ruleset

File: overlay.yaml [hunk 3]
```diff
@@ -0,0 +1,1057 @@
+overlay: 1.0.0
```
Comment (TooTallNate) [inline overlay.yaml]: Does this file need to be at the root of the repo? Why not inside `packages/sdk`?

Comment (ismaelrumzan) [reply overlay.yaml]: Good catch - updated

**Review by EndangeredMassa — DISMISSED** [review]

**Review by molebox — DISMISSED** [review]

**Review by molebox — APPROVED** [review]

**Pull request merged**